File: src/app/src/main/java/org/datn/app/core/service/OrderService.java
package org.datn.app.core.service;

import org.datn.app.core.dto.OrderRequest;
import org.datn.app.core.dto.OrderStatisticalDto;
import org.datn.app.core.entity.Order;
import org.datn.app.core.service.base.ServiceCore;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;

import java.util.Map;

public interface OrderService extends ServiceCore<Order, Long> {
    Order findByCode(String code);

    Order findByCodeAndStatus(String code, String status);

    Map<String, Object> doOrder(OrderRequest model);

    ResponseEntity<?> approveOrder(Long id);

    ResponseEntity<?> cancelOrder(Long id);

    // hoàn hàng
    ResponseEntity<?> returnOrder(Long id,String note);

    Page<Order> findAllByUserId(Long userId, Pageable pageable);

    Page<Order> findAll(Integer page, Integer size);

    // thống kê
    Page<Order> findAllByStatus(OrderStatisticalDto model);

    // giao hàng
    ResponseEntity<?> deliveryOrder(Long id);

    // nhận hàng
    ResponseEntity<?> receiveOrder(Long id);

    // xem chi tiết đơn hàng
    ResponseEntity<?> detailOrder(Long id);
}

File: src/user-ui/src/core/service/order.service.ts
import {OrderRequest} from "@/core/model/order.model";
import axios from "axios";
import {toast} from "vue3-toastify";

export class OrderService {
    url = "/api/order/";

    async doOrder(model: OrderRequest) {
        const response = await axios.post(this.url + "doOrder", model);
        if (response.data?.status == 200) {
            toast.success(response.data?.message);
        }else if(response.data.status == 500){
            toast.error(response.data?.message);
        }
    }

    async getPrice(model: OrderRequest) {
        const response = await axios.post(  "/api/cart/getPriceByCartIdList", model);
        return response.data;
    }

    async getOrderByUserId(userId : number, page : number, size :number) {
        const response = await axios.get(this.url + "findAllByUserId/" + userId + "/" + page + "/" + size);
        return response.data;
    }

    async cancelOrder(id : number) {
        const response = await axios.post(this.url + "cancel/" + id);
        if (response.data?.status == 200) {
            toast.success(response.data?.message);
            return true;
        }else if(response.data.status == 500){
            toast.error(response.data?.message);
            return false;
        }
    }

    async returnOrder(id : number,note: string) {
        const response = await axios.post(this.url + "return/" + id + "/" + note);
        if (response.data?.status == 200) {
            toast.success(response.data?.message);
            return true;
        }else if(response.data.status == 500){
            toast.error(response.data?.message);
            return false;
        }
    }

    async receiveOrder(id : number) {
        const response = await axios.post(this.url + "receive/" + id);
        if (response.data?.status == 200) {
            toast.success(response.data?.message);
            return true;
        }else if(response.data.status == 500){
            toast.error(response.data?.message);
            return false;
        }
    }
}